src/transxchange/read.rs

```diff
@@ -168,15 +187,100 @@ fn load_companies(transxchange: &Element) -> Result<CollectionWithId<Company>> {
     Ok(companies)
 }
 
+fn load_commercial_physical_modes(
+    transxchange: &Element,
+) -> Result<(CommercialMode, PhysicalMode)> {
+    let mode = match transxchange
+        .try_only_child("Services")?
+        .try_only_child("Service")?
+        .try_only_child("Mode")
+        .map(Element::text)
+    {
+        Ok(mode) => MODES.get(mode.as_str()).unwrap_or(&DEFAULT_MODE),
+        Err(e) => {
+            warn!("{}", e);
+            DEFAULT_MODE
+        }
+    };
+    let commercial_mode = CommercialMode {
+        id: mode.to_string(),
+        name: mode.to_string(),
+    };
+    let physical_mode = PhysicalMode {
+        id: mode.to_string(),
+        name: mode.to_string(),
+        ..Default::default()
+    };
+    Ok((commercial_mode, physical_mode))
+}
+
+fn load_lines(
+    transxchange: &Element,
+    network_id: &str,
+    commercial_mode_id: &str,
+) -> Result<CollectionWithId<Line>> {
+    let service = transxchange
+        .try_only_child("Services")?
+        .try_only_child("Service")?;
+    let service_id = service.try_only_child("ServiceCode")?.text();
+    let mut lines = CollectionWithId::default();
+    for line in service.try_only_child("Lines")?.children() {
+        if let Some(line_id) = line.attr("id") {
+            let id = format!("{}:{}", service_id, line_id);
+            let code = Some(line.try_only_child("LineName")?.text().trim().to_string());
+            let name = if let Ok(description) = service.try_only_child("Description") {
```

Comment:
name, forward_name, backward_name, network_id and commercial_mode_id 
can be out of the loop (if you want)

Reply:
Good point. Done in 2b710d24a4b4bea5ae6fc692c331abb5b37db960.